aten/src/ATen/native/BinaryOps.cpp

```diff
@@ -101,6 +101,8 @@ Tensor& mul_(Tensor& self, const Tensor& other) {
 }
 
 Tensor& sub_out(Tensor& result, const Tensor& self, const Tensor& other, Scalar alpha) {
+  TORCH_CHECK(self.scalar_type() != kBool && other.scalar_type() != kBool,
+              "Subtraction, the `-` operator, on bool tensors is not supported. Use the `^` operator instead.");
```

Comment:
Sorry, one last thing, can you, please, update the message to: "Subtraction, the `-` operator, on bool tensors is not supported. Use the `~` or '.bitwise_not()' operator instead."

Consider the case of 1 - boolTensor, '^' wont work here.

Reply:
I thought `~` hasn't been implemented on bool tensors yet? I've added `bitwise_not()` for now. Let me know if you wanna add `~` as well!

Reply:
You are right. Your pr was reverted as it had to go with some other set of changes, you can see your changes in this [PR](https://github.com/pytorch/pytorch/pull/21113) which will be landing very soon so i guess its ok to set this message.

Reply:
Is your PR gonna be merged on Monday? I'm a bit worried because there will be a release cut on Monday and this message can be confusing if you PR will not be merged by then.

Reply:
Yeah, it definitely will.

Reply:
OK, good to know! Updated!